app/service/rest_svc.py

```diff
@@ -187,6 +187,16 @@ def __init__(self):
         operation = (await self.get_service('data_svc').locate('operations', match=dict(id=link.operation)))[0]
         return await operation.apply(link)
 
+    async def task_agent_with_ability(self, paw=None, ability_id=None):
+        abilities = await self.get_service('data_svc').locate('abilities', match=dict(ability_id=ability_id))
+        agent = await self.get_service('data_svc').locate('agents', match=dict(paw=paw))
```

Comment:
this is way better. 

last thing: instead of defining agent here and then finding it in the operation below, can we just do the operation loop up here (returning an operation object, not an agent).

Reply:
I still need the specific agent object... so I can, but I still need a specific agent object from the operation.